kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ActionWrapper.java

```diff
@@ -15,29 +15,89 @@
  */
 package io.delta.kernel.internal.replay;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import io.delta.kernel.data.ColumnVector;
 import io.delta.kernel.data.ColumnarBatch;
+import io.delta.kernel.utils.FileStatus;
+import io.delta.kernel.internal.checkpoints.CheckpointInstance;
+import io.delta.kernel.internal.checkpoints.SidecarFile;
+import io.delta.kernel.internal.fs.Path;
+import io.delta.kernel.internal.util.FileNames;
 
 /** Internal wrapper class holding information needed to perform log replay. */
 class ActionWrapper {
-    private final ColumnarBatch columnarBatch;
-    private final boolean isFromCheckpoint;
+    private ColumnarBatch columnarBatch;
+    private final Path checkpointPath;
     private final long version;
 
-    ActionWrapper(ColumnarBatch data, boolean isFromCheckpoint, long version) {
+    private boolean containsSidecarsInSchema;
+
+    ActionWrapper(
+            ColumnarBatch data,
+            Path checkpointPath,
+            long version,
+            boolean containsSidecarsInSchema) {
         this.columnarBatch = data;
-        this.isFromCheckpoint = isFromCheckpoint;
+        this.checkpointPath = checkpointPath;
         this.version = version;
+        this.containsSidecarsInSchema = containsSidecarsInSchema;
     }
 
     public ColumnarBatch getColumnarBatch() {
         return columnarBatch;
     }
 
     public boolean isFromCheckpoint() {
-        return isFromCheckpoint;
+        return checkpointPath != null;
     }
 
     public long getVersion() {
         return version;
     }
+
+    public boolean getContainsSidecarsInSchema() {
+        return containsSidecarsInSchema;
+    }
+
+    /**
+     * Reads SidecarFile actions from ColumnarBatch, appending the SidecarFile actions to the end
```

Comment:
update the description as we are not adding to the end of the list anumore.

Reply:
Fixed.